tasks.py

```diff
@@ -541,6 +541,73 @@ def generate_password(c, user="root"):
     print(f"{user}-password: {passw}")
     print(f"{user}-password-hash: {out.stdout}")
 
+@task
+def add_server(c, hostname):
+    """
+    Generate new server keys and configurations for a given hostname and hardware config
+    """
+
+    import subprocess
+
+    print(f"Adding {hostname}")
+
+    sopsPermissions = None
+    with open("sopsPermissions.json","r") as f:
+        sopsPermissions = f.read()
+    sopsPermissions = json.loads(sopsPermissions)
+    if(sopsPermissions.get(f"hosts/{hostname}.yml$",None)):
+       print("Configuration already exists")
+       exit(-1)
+    sopsPermissions[f"hosts/{hostname}.yml$"] = []
+    
+    
+    with open("sopsPermissions.json","w") as f:
+        json.dump(sopsPermissions,f)
+    update_sops_files(c)
+
+    sops_file = f"{ROOT}/hosts/{hostname}.yml"
+
+    #Generate the password
+    print("Generating Password")
+    size = 12
+    chars = string.ascii_letters + string.digits
+    passwd = "".join(random.choice(chars) for x in range(size))
+    passwd_str = subprocess.Popen(["echo",f"{passwd}"],stdout=subprocess.PIPE,text=True)
+    passwd_hash = subprocess.check_output(["mkpasswd", "-m", "sha-512", "-s"], stdin=passwd_str.stdout,text=True)
```

Comment:
No need for a pipe here.

```suggestion
    passwd_hash = subprocess.check_output(["mkpasswd", "-m", "sha-512", "-s"], input=passwd,text=True)
```